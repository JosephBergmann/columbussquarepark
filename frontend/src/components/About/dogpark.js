import React, { useState, Fragment } from 'react';
import { useAccessibilitySettings } from '../../context/accessibility';


export default function DogPark() {
    const { accessibilitySettings } = useAccessibilitySettings();
    const { darkMode, textSize } = accessibilitySettings;

    return (
        <div className="py-6 px-8">
            <h1 className={`${darkMode && "text-white"} ${textSize ? "text-2xl" : "text-xl"} underline leading-6 pb-2 mt-2`}>Dog Park Rules</h1>
            <p className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} my-4`}>Failure to abide by the park rules may result in loss of privileges. All visitors must be accompanied by a dog. Owners are fully responsible for the conduct of their dog(s).</p>
            <p className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} my-4`}>By entering the dog run, you agree to the following rules:</p>
            <ol className="list-decimal py-2 px-10">
                <li className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} mb-4`}>Aggressive behavior will not be tolerated from dogs or owners. Dogs must be removed at the first sign of aggression.</li>

                <li className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} mb-4`}>The following are prohibited in the park:
                    <ul className="list-disc pl-8">
                        <li>puppies under 4 months old,</li>
                        <li>dogs with contagious disease,</li>
                        <li>adult dogs that are not spayed or neutered.</li>
                    </ul>
                </li>

                <li className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} mb-4`}>Handlers must maintain vocal and visual control over their dog(s).</li>

                <li className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} mb-4`}>Food of any kind is prohibited. Small training treats are permitted when concealed and used responsibly.</li>

                <li className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} mb-4`}>Unplug and Engage with your dog. For your dog's safety and enjoyment, please keep your phone and other electronic devices away while in the park. Engage and interact with your furry friend and other park-goers to create a positive social experience for all. Checking lottery results can wait until after your visit, but your dog's safety and happiness cannot.</li>

                <li className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} mb-4`}>Three dogs maximum per handler. Handlers must remain in the run.</li>

                <li className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} mb-4`}>Dogs in the run must be off leash, and dogs outside of the run must be on leash.</li>

                <li className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} mb-4`}>All handlers must clean up the dog waste from their dog(s).</li>

                <li className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} mb-4`}>Dogs must be licensed and vaccinated; tags must be worn at all times.</li>

                <li className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} mb-4`}>Dog owners must be 18 years of age to enter. Anyone between 14 and 17 must be accompanied by an adult. No children under the age of 14 are permitted in the dog run.</li>
            </ol>
        </div>
    )
}
