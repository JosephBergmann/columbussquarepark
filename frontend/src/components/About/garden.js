import React, { useState, Fragment } from 'react';
import { useAccessibilitySettings } from '../../context/accessibility';


export default function GardenPolicy() {
    const { accessibilitySettings } = useAccessibilitySettings();
    const { darkMode, textSize } = accessibilitySettings;

    return (
        <div className='py-6 px-8'>
            <h1 className={`${darkMode && "text-white"} ${textSize ? "text-2xl" : "text-xl"} underline leading-6 pb-2 mt-2`}>Garden Policy</h1>
            <p className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} my-4`}>
                All gardens installed by volunteers, partner organizations or contractors must submit detailed plans that include a site design, plant species, maintenance plan, funding sources and installation timelines. Plans must be submitted to the appropriate Operations Manager and, if applicable, the Facility Supervisor. Applications will be Ô¨Årst vetted at regional meetings and, if they are approved, the plant list will be reviewed by an Operations Manager. Once the application and plant list are approved the applicant will be given a Parks & Recreation work Permit, and can begin to install the garden.
            </p>
            <p className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} my-4`}>
                Safety will remain top priority in reviewing the plans. In particular, there must be assurances the proposed plants will not obstruct sightlines into the park. PP&R will also consider if new planting will encroach on other activities at the site. You must avoid planting anything that will grow into a fence, a gate or close to building foundations that can cause property damage. Do not plant anything that will overhang into pathways or sidewalks. Do not plant thorny plants near public walkways or any highly trafficked area that could potentially injure children or people using the space. Do not plant invasive species. Avoid plants that will spread outside of boundaries of the designated garden bed.
            </p>
            <p className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} my-4`}>
                The planting design will be evaluated based on the overall best use for the space and with consideration for neighbors and other user groups in mind.
            </p>
            <p className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} my-4`}>
                All plant material, once planted, becomes the property of the City of Philadelphia, under the jurisdiction of Philadelphia Parks and Recreation and therefore the department will dictate whether it remains or is removed over time.
            </p>
            <p className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} my-4`}>
                The plans must be legible and must include specific measurements, navigational coordinates and a plant list.
            </p>
        </div>
    )
};
