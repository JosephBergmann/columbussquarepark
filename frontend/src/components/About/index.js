import React, { useState, Fragment } from 'react';
import { Dialog, Transition, Switch, Tab } from '@headlessui/react';
import { useAccessibilitySettings } from '../../context/accessibility';
import DogPark from './dogpark';
import GardenPolicy from './garden';
import cspMap from './csp-map.jpg';


// aboutView = ['doge', 'garden', 'art']

export default function About() {
    const { accessibilitySettings } = useAccessibilitySettings();
    const { darkMode, textSize } = accessibilitySettings;

    const [aboutView, setAboutView] = useState('map')

    return (
        <div className="mt-6 mb-20">
            <div className="my-4 mb-10">
                <h1 className={`${darkMode && "text-white"} ${textSize ? "text-3xl" : "text-2xl"} leading-6 pb-4`}>About</h1>
                <p className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} mt-4`}>Columbus Square Park is a vibrant community gathering place located between 12th and 13th Streets and Wharton and Reed streets in the Passyunk Square neighborhood in South Philadelphia.</p>
                <p className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} mt-4`}>Columbus Square provides a diverse array of amenities for both residents and visitors. Whether you're a dog lover, a parent with young children, a sports enthusiast, or simply someone who appreciates a moment of tranquility, Columbus Square caters to all. Explore our rose garden and meandering nature walk that encircles the park, offering a delightful experience for nature enthusiasts. Serving as an oasis from the urban hustle and bustle, Columbus Square is a dynamic community space that continues to evolve.</p>
                <p className={`${darkMode && "text-white"} ${textSize ? "text-lg" : null} mt-4`}>Come experience Columbus Square for yourself. Volunteer and GET ENGAGED!</p>
            </div>
            <div>
                <Tab.Group>
                    <Tab.List className="grid grid-cols-3 bg-primary rounded-xl">
                     <Tab as={Fragment}>
                            {({ selected}) => (
                                    <button
                                        className={`md:px-4 py-2 m-1 rounded-lg text-lg ${selected ? 'bg-fun' : 'bg-primary hover:bg-secondary'} ${textSize && "text-xl"}`}
                                        onClick={() => setAboutView('map')}
                                    >
                                        Map
                                    </button>
                            )}
                        </Tab>
                        <Tab as={Fragment}>
                            {({ selected}) => (
                                    <button
                                        className={`md:px-4 py-2 m-1 rounded-lg text-lg ${selected ? 'bg-fun' : 'bg-primary hover:bg-secondary'} ${textSize && "text-xl"}`}
                                        onClick={() => setAboutView('dog')}
                                    >
                                        Dog Park Rules
                                    </button>
                            )}
                        </Tab>
                        <Tab as={Fragment}>
                            {({ selected}) => (
                                    <button
                                        className={`md:px-4 py-2 m-1 rounded-lg text-lg ${selected ? 'bg-fun' : 'bg-primary hover:bg-secondary'} ${textSize && "text-xl"}`}
                                        onClick={() => setAboutView('garden')}
                                    >
                                        Garden Policy
                                    </button>
                            )}
                        </Tab>
                    </Tab.List>
                </Tab.Group>
                <div className={`rounded-xl bg-slate-200 mt-4 mb-10 ${darkMode && "bg-gray-800"}`}>
                    {aboutView === 'dog' && <DogPark />}
                    {aboutView === 'garden' && <GardenPolicy />}
                    {aboutView === 'map' && <div className='p-8 flex justify-center'>
                        <img src={cspMap} />
                    </div>}
                </div>
            </div>
        </div>
    )
}
