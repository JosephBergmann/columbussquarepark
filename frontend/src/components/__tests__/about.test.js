import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import About from '../About'

test('should render about component', () => {
    render(<About />)
    const homeElement = screen.getByTestId('about-1');
    expect(homeElement).toBeInTheDocument();
})

test('matches snapshot', () => {
    const tree = renderer.create(<About />).toJSON();
    expect(tree).toMatchSnapshot();
})
