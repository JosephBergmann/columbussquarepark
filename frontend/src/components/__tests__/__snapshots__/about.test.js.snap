// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="about-1"
>
  <p
    className="mb-5"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
  </p>
  <p
    className="mb-5"
  >
    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>
  <p
    className="mb-5"
  >
    Nullam sit amet purus ut erat dignissim congue. Fusce convallis hendrerit orci, id pulvinar odio volutpat eget. Proin consequat cursus tellus, ut fermentum est bibendum vel.
  </p>
</div>
`;
